### PR TITLE
DrivAerML: weight decay sweep on champion config

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The DM champion config (val=3.833%, TEST=4.324% full-eval) uses **NO weight decay**. WD=1e-2 was a critical component in both TandemFoil (all winners use WD=1e-2) and AirfRANS (WD=1e-2 + gc compound). On DM, we've never tested WD on the champion. 

Seed=7 recovery (PR #3286) confirmed the champion config ceiling is ~3.83% val regardless of initialization — breaking past this ceiling requires a config change. WD regularization could reduce overfitting on the 400-sample DM training set and lower the test metric gap (currently 4.324% TEST vs 3.833% val = 0.49pp generalization gap).

This sweep tests WD=1e-3 and WD=1e-2 against the no-WD champion to find if weight decay improves DM test generalization.

## Instructions

Run a 2-trial sweep: WD=1e-3 (moderate) and WD=1e-2 (full, matching TF/AF). Use `--wandb-group dm-weight-decay` to group runs.

**Trial 1 — WD=1e-3 (moderate):**
```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine-t-max 30 \
  --grad-clip 0.5 \
  --weight-decay 1e-3 \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --ema-decay 0.9995 \
  --ema-mode fixed \
  --no-compile-model \
  --save-checkpoint \
  --eval-interval 10 \
  --seed 42 \
  --wandb-group dm-weight-decay \
  --kill-if-best-val-above 150:8.0 \
  --kill-if-no-improvement 250:0.05
```

**Trial 2 — WD=1e-2 (strong, matching TF/AF):**
```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine-t-max 30 \
  --grad-clip 0.5 \
  --weight-decay 1e-2 \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --ema-decay 0.9995 \
  --ema-mode fixed \
  --no-compile-model \
  --save-checkpoint \
  --eval-interval 10 \
  --seed 42 \
  --wandb-group dm-weight-decay \
  --kill-if-best-val-above 150:8.0 \
  --kill-if-no-improvement 250:0.05
```

Run trials **sequentially** (Trial 1 first, then Trial 2). If Trial 1 beats baseline, note it but still run Trial 2 to find the better WD value.

**After training**, run full-eval on the best checkpoint from the better trial:
```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --load-checkpoint <path-to-best-checkpoint> \
  --eval-only \
  --optimizer adamw --lr 5e-4 --cosine-t-max 30 --grad-clip 0.5 \
  --weight-decay <best-wd> --enable-fourier --model-layers 4 \
  --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 --ema-decay 0.9995 \
  --ema-mode fixed --no-compile-model
```

## Baseline

- **DM best val:** 3.833% at ep511 (PR #3072, no WD, W&B: ncl1dh88)
- **DM authoritative full-eval TEST:** 4.324% (faye run zx5syby1)
- **DM batch-limited TEST:** 4.685%
- **Val-test gap:** 0.49pp (suggests overfitting that WD may reduce)
- **External SOTA:** <3.71% (AB-UPT)
- **WD evidence from other benchmarks:** TF uses WD=1e-2 in all champions; AF uses WD=1e-2 + gc compound

## Expected output

Report per trial:
1. Best val `surface_rel_l2_pct` and epoch
2. W&B run ID
3. Val-test gap comparison (does WD reduce the 0.49pp gap?)
4. Full-eval TEST from best checkpoint

### CODE REVIEW

**Comment by morganmcg1** [comment]
HUMAN: Closing during the final cost-control harvest pass. The associated student pod (megumi) was intentionally stopped after its .claude logs were archived locally, and this PR was not in the small set kept running for final test-metric harvesting. Reason: this weight-decay branch was secondary to the kept DrivAerML schedule/LR/test-metric candidates in the final cost-cut pass.
